rustls/src/common_state.rs

```diff
@@ -273,6 +273,27 @@ impl CommonState {
         Ok(self.write_fragments(outgoing_tls, fragments))
     }
 
+    pub(crate) fn validate_rpk_negotiation(
+        &mut self,
+        local_expects_rpk: bool,
+        peer_supports_rpk: bool,
```

Comment:
I'm a bit nervous about this API signature that takes two `bool` values. It seems like it would be easy to mistakenly switch the order around. Perhaps a dedicated `enum` that covers both sides of this unambiguously would be in order?

Reply:
Yeah that is tricky indeed. I've added a struct `RawKeyNegotationParams`, with fields 'local_expects_rpk` and `peer_supports_rpk` as fields. It should be harder to mistakenly switch the order around now.